snapcraft/projects.py

```diff
@@ -547,6 +548,16 @@ def _validate_architecture_data(cls, architectures):
         """Validate architecture data."""
         return _validate_architectures(architectures)
 
+    @pydantic.validator("provenance")
+    @classmethod
+    def _validate_provenance(cls, provenance):
+        if provenance and not re.match(r"^[a-zA-Z0-9-]+$", provenance):
+            raise ValueError(
+                "provenance must consist of alphanumeric characters and hyphens."
```

Comment:
maybe "alphanumeric characters and/or hyphens"?

Reply:
Good idea, that's more accurate.